pkg/parser/charset/encoding_base.go

```diff
@@ -123,12 +122,10 @@ func generateEncodingErr(name string, invalidBytes []byte) error {
 // HackSlice converts string to slice without copy.
 // Use at your own risk.
 func HackSlice(s string) (b []byte) {
```

Comment:
`nil` `[]byte` and zero length `[]byte` may have different behaviour at caller. Let's check CI

Reply:
will dig it tomorrow~